packages/core/src/Popover/Popover.tsx

```diff
@@ -0,0 +1,165 @@
+import * as Popover from '@radix-ui/react-popover'
+import React, { useEffect, useRef } from 'react'
+import { SpaceProps } from 'styled-system'
+import { useTheme } from 'styled-components'
+import { Box, BoxProps } from '../Box/Box'
+import { CloseButton } from '../CloseButton/CloseButton'
+import { ThemeProvider } from '../ThemeProvider/ThemeProvider'
+import Overlay from './Overlay'
+import { FixedCloseButton, StyledArrow } from './Popover.styles'
+
+export type PopoverProps = {
+  renderContent: () => React.ReactNode
+  idx: string
+  isOpen?: boolean
+  color?: string
+  children?: React.ReactNode
+  hideArrow?: boolean
+  hideOverlay?: boolean
+  onOpen?: () => void
+  onClose?: () => void
+  openOnHover?: boolean
+  borderRadius?: BoxProps['borderRadius']
+  width?: string | number
+  trapFocus?: boolean
+  toggleIsOpenOnClick?: boolean
+  openOnMount?: boolean
+  overlayOpacity?: number
+  p?: SpaceProps['p']
+  placement?:
+    | 'top'
+    | 'top-start'
+    | 'top-end'
+    | 'bottom'
+    | 'bottom-start'
+    | 'bottom-end'
+    | 'left'
+    | 'left-start'
+    | 'left-end'
+    | 'right'
+    | 'right-start'
+    | 'right-end'
+}
+
+function parsePlacement(placement: PopoverProps['placement']): {
+  side: 'top' | 'bottom' | 'left' | 'right'
+  align: 'start' | 'end' | 'center'
+} {
+  const match = /([a-z]+)-?([a-z]+)?/.exec(placement)
+  if (match) {
```

Comment:
if you add Array.isArray and a length check, can we drop ts-ignore?

Reply:
The issue is with the parts of the string we're trying to extract. There's no guarantee that the `string` that's extracted matches the expected return values.

![CleanShot 2024-07-19 at 08 52 49@2x](https://github.com/user-attachments/assets/5c68dd25-fb87-4ee4-ad7a-691c8afeb0ec)